swift/ql/src/queries/Security/CWE-328/WeakSensitiveDataHashing.qhelp

```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE qhelp PUBLIC
+        "-//Semmle//qhelp//EN"
+        "qhelp.dtd">
+<qhelp>
+    <overview>
+        <p>
+            Using a broken or weak cryptographic hash function can leave data
+            vulnerable, and should not be used in security related code.
+        </p>
+
+        <p>
+            A strong cryptographic hash function should be resistant to:
+        </p>
+        <ul>
+            <li>
+                pre-image attacks: if you know a hash value <code>h(x)</code>,
+                you should not be able to easily find the input <code>x</code>.
+            </li>
+            <li>
+                collision attacks: if you know a hash value <code>h(x)</code>,
+                you should not be able to easily find a different input
+                <code>y</code>
+                with the same hash value <code>h(x) = h(y)</code>.
+            </li>
+        </ul>
+        <!--<p>
+             In cases with a limited input space, such as for passwords, the hash
+             function also needs to be computationally expensive to be resistant to
+             brute-force attacks. Passwords should also have an unique salt applied
+             before hashing, but that is not considered by this query.
+        </p>-->
+
+        <p>
+            As an example, both MD5 and SHA-1 are known to be vulnerable to collision attacks.
+        </p>
+
+        <p>
+            Since it's OK to use a weak cryptographic hash function in a non-security
+            context, this query only alerts when these are used to hash sensitive
+            data (such as passwords, certificates, usernames).
+        </p>
+
+    </overview>
+    <recommendation>
+
+        <p>
+            Ensure that you use a strong, modern cryptographic hash function:
+        </p>
+
+        <ul>
+            <li>
+                such as Argon2, scrypt, bcrypt, or PBKDF2 for passwords and other data with limited input space.
```

Comment:
It would be good to clarify the phrase "limited input space" somewhere in this doc.